library/general/src/lib/yast2/group.rb

```diff
@@ -0,0 +1,49 @@
+require "yast2/execute"
+
+module Yast2
+  class Group
+    class << self
+      GETENT_GROUP_MAPPING = {
+        "name" => 0,
+        "passwd" => 1,
+        "gid" => 2,
+        "users" => 3
+      }
+
+      def all
+        @all ||= read
+      end
+
+      def reset
+        @all = nil
+      end
+
+      def read
+        getent = Yast::Execute.on_target!("/usr/bin/getent", "group", stdout: :capture)
+        getent.lines.map do |line|
+          values = line.chomp.split(":")
+          new(
+            values[GETENT_GROUP_MAPPING["name"]],
+            gid: values[GETENT_GROUP_MAPPING["gid"]],
+            users_name: values[GETENT_GROUP_MAPPING["users"]]&.split(",") || []
+          )
+        end
+      end
+    end
+
+    attr_reader :name, :gid, :users_name
+
+    def initialize(name, gid: nil, users_name: [])
+      @name = name
+      @gid = gid
+      @users_name = users_name
+    end
+
+    def users
+      # lazy load to avoid circular deps
+      require "yast2/user"
+      User.all.select { |u| u.gid == gid || users_name.include?(u.name) }
```

Comment:
Bear in mind that you can only have a single representation of users/groups. For instance, it is not possible to have a separate representation of "current" and "target" states. I know, perhaps it is not useful, but I kind of like that approach that we followed in storage-ng and network-ng.

Reply:
Additionally, what about having a `UsersCollection` class? It could define powerful query methods. Perhaps it is not that important at that point, but I just wonder whether it might be useful.